### PR TITLE
add setTransactionMessageLifetimeUsingBlockhash

### PR DESCRIPTION
**Pull request by mcintyre94**
Implementation + tests are from @solana/transactions

### CODE REVIEW

**Comment by mcintyre94** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2426?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2426** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2425** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2425?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2424** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2424?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2423** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2423?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mcintyre94 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/transaction-messages/src/blockhash.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+export function assertIsTransactionMessageWithBlockhashLifetime(
+    transaction: BaseTransactionMessage | (BaseTransactionMessage & ITransactionMessageWithBlockhashLifetime),
+): asserts transaction is BaseTransactionMessage & ITransactionMessageWithBlockhashLifetime {
+    if (!isTransactionMessageWithBlockhashLifetime(transaction)) {
+        throw new SolanaError(SOLANA_ERROR__TRANSACTION__EXPECTED_BLOCKHASH_LIFETIME);
+    }
+}
+
+export function setTransactionMessageLifetimeUsingBlockhash<
```
Comment (steveluscher) [inline packages/transaction-messages/src/blockhash.ts:40]: People are just absolutely going to hate me for this.

The method name really does explain what's _actually_ happening though.

**Review by steveluscher — APPROVED** [review]
Added via Giphy

**Review by lorisleiva — APPROVED** [review]

File: packages/transaction-messages/src/blockhash.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/transaction-messages/src/blockhash.ts:40]: I don't think it's too bad tbh. 🙂

**Pull request merged**